nixos/modules/config/update-users-groups.pl

```diff
@@ -213,7 +219,11 @@ sub parseUser {
 
     # Create a home directory.
     if ($u->{createHome}) {
-        make_path($u->{home}, { mode => 0700 }) if ! -e $u->{home};
+        if (! -e $u->{home}) {
+             my @dirs = splitdir($u->{home});
+             make_path(catdir(@dirs[0..$#dirs - 1]), { mode => 0755 });
```

Comment:
Wouldn't it be simpler to only call `make_path` on the last component? That functions seems to create all intermediate directories already